cmd/juju/application/upgradecharm.go

```diff
@@ -747,16 +747,20 @@ func allEndpoints(ci *charms.CharmInfo) set.Strings {
 	return epSet
 }
 
+// detectDefaultSpace returns the current default space for the bindings provided.
+// The default space is determined via the following sequence.
+// 1. the space of the default endpoint in the bindings provided.
+// 2. the "default-space" from the model-config.
+// 3. AlphaSpaceName
 func detectDefaultSpace(modelConfig *config.Config, curBindings map[string]string) string {
```

Comment:
I wonder whether this can somehow be moved (in a future PR) to one of the core packages so we can share it between client and controller...

Reply:
👍

Reply:
Turns out this is not required anymore. I have an upcoming PR to strip this out

Reply:
applied the patches from @achilleasa to do this, instead of a new PR